http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyConfiguration.java

```diff
@@ -1,103 +0,0 @@
-/*
- * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- * You may not use this file except in compliance with the License.
- * A copy of the License is located at
- *
- *  http://aws.amazon.com/apache2.0
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
-
-package software.amazon.awssdk.http.nio.netty.internal;
-
-import static software.amazon.awssdk.http.SdkHttpConfigurationOption.CONNECTION_TIMEOUT;
-import static software.amazon.awssdk.http.SdkHttpConfigurationOption.MAX_CONNECTIONS;
-import static software.amazon.awssdk.http.SdkHttpConfigurationOption.USE_STRICT_HOSTNAME_VERIFICATION;
-import static software.amazon.awssdk.utils.NumericUtils.saturatedCast;
-
-import java.time.Duration;
-import java.util.Optional;
-import software.amazon.awssdk.annotations.Immutable;
-import software.amazon.awssdk.annotations.ReviewBeforeRelease;
-import software.amazon.awssdk.http.nio.netty.NettySdkHttpClientFactory;
-import software.amazon.awssdk.utils.AttributeMap;
-
-/**
- * Helper class to unwrap and convert service defaults.
- */
-@Immutable
-public final class NettyConfiguration {
```

Comment:
Why get rid of this? It's is a nice facade over AttributeMap

Reply:
It just felt unnecessary because everything was in the map now. We can add it back in...

Reply:
`NettyConfiguration` is an internal class. Customers don't see it. The map is only used in one method outside of its originating class, so it really feels like overkill.

Reply:
I'm also using it in my H2 classes. Lets add it back.